File: Python3/1678. Goal Parser Interpretation.py
class Solution:
    def interpret(self, command: str) -> str:
        ans = ''
        i = 0
        
        while i < len(command):
            if command[i] == 'G':
                ans += 'G'
            elif command[i] == '(':
                if command[i+1] == ')':
                    ans += 'o'
                    # increment an extra step
                    i += 1
                else:
                    ans += 'al'
                    # increment extra three steps
                    i += 3
            i += 1
            
        return ans
                    

File: Java/1678. Goal Parser Interpretation.java
class Solution {
    public String interpret(String command) {
        StringBuilder ans = new StringBuilder();
        int i = 0;
        
        while (i < command.length()) {
            if (command.charAt(i) == 'G') {
                ans.append('G');
            } else if (command.charAt(i) == '(') {
                if (command.charAt(i+1) == ')') {
                    ans.append('o');
                    i++;
                } else {
                    ans.append("al");
                    i += 3;
                }
            }
            i++;
        }
        
        return ans.toString();
    }
}